gulp-settings/aws-lambda.js

```diff
@@ -0,0 +1,124 @@
+const gulp          = require('gulp');
+const gutil         = require('gulp-util');
+const del           = require('del');
+const install       = require('gulp-install');
+const zip           = require('gulp-zip');
+const AWS           = require('aws-sdk');
+const fs            = require('fs');
+const path          = require('path');
+const merge         = require('merge-stream');
+const sequence      = require('gulp-sequence');
+const changeCase    = require('change-case');
+const minimist      = require('minimist');
+const _             = require('lodash');
+
+const LAMBDA_RELATIVE_DIR = './aws/lambdas';
+const LAMBDA_ABOSULTE_DIR = path.join(path.resolve(), LAMBDA_RELATIVE_DIR);
+const lambdaFolders = getDirectories(LAMBDA_RELATIVE_DIR);
+
+const { src: commandLineLambdas = 'all' } = minimist(process.argv.slice(2));
+
+function getDirectories(sourcePath) {
+    return fs.readdirSync(sourcePath).filter(file => {
+        return fs.statSync(path.join(path.resolve(), sourcePath, file)).isDirectory();
+    });
+}
+
+gulp.task('aws-lambda', sequence(
+    'clean:lambda:zip',
+    'npm-dependencies',
+    'zip',
+    'upload'
+));
+
+gulp.task('clean:lambda:zip', () => {
+    const zipFiles = lambdaFolders.map(lambda => {
+        return path.join(LAMBDA_ABOSULTE_DIR, lambda, 'build.zip');
+    });
+
+    return del.sync(zipFiles);
+});
+
+gulp.task('npm-dependencies', () => { 
+    const gulpSources = lambdaFolders.map(lambda => {
+        return gulp.src(path.join(LAMBDA_ABOSULTE_DIR, lambda , 'package.json'))
+            .pipe(gulp.dest(`${LAMBDA_RELATIVE_DIR}/${lambda}`))
+            .pipe(install({ production: true }));
+    });
+
+    return merge.apply(null, gulpSources);
+});
+
+gulp.task('zip', () => {
+    const gulpSources = lambdaFolders.map(lambda => {
```

Comment:
I think this should be named differently because it is a bit misleading. Although I don't really know what it is supposed to be, I guess lambda builds maybe?